docs/sources/lids/_DRAFT_RemoteRuleEvaluation.md

```diff
@@ -0,0 +1,84 @@
+---
+title: "XXXX: Remote Rule Evaluation"
+---
+
+# Remote Rule Evaluation
+
+**Author:** Danny Kopping (danny.kopping@grafana.com)
+
+**Date:** 01/2023
+
+**Sponsor(s):** @dannykopping
+
+**Type:** Feature
+
+**Status:** Draft
+
+**Related issues/PRs:** https://github.com/grafana/mimir/pull/1536
+
+**Thread from [mailing list](https://groups.google.com/forum/#!forum/lokiproject):** N/A
+
+---
+
+## Background
+
+The `ruler` is a component that evaluates alerting and recording rules. Loki reuses Prometheus' rule evaluation engine. The `ruler` currently operates by initialising a `querier` internally and evaluating all rules "locally" (i.e. it does not rely on any other components). Each rule group executes concurrently, and rules within the rule group are evaluated sequentially (this is an implementation detail from Prometheus).
+
+Recording rules produce metric series which are sent to a Prometheus-compatible source. Alerting rules send notifications to Alertmanager when a condition is met. Both of these rule types can play a vital role in an organisation's observability strategy, and so their reliable evaluation is essential.
+
+## Problem Statement
+
+Rule evaluations can contain expensive queries. The `ruler` initialises a `querier`, but the `querier` does not have the capability to accelerate queries; the `query-frontend` component is responsible for query acceleration through splitting, sharding, caching, and other techniques.
+
+An expensive rule query can cause an entire `ruler` instance to use excessive resources and even crash by running out of memory. This is highly problematic for the following reasons:
+
+- slow rule evaluations can lead to subsequent rules in a group to be delayed or missed, leading to missing alerts or gaps in recording rule metrics
+- excessive resource usage can impede the evaluation of rules for other tenants (noisy neighbour)
+
+## Goals
+
+- faster, more efficient rule evaluation
+- greater isolation between tenants
+- more reliable service
+
+## Non-Goals
+
+This proposal does not aim to make this option the default mode of evaluation; it should be optional because it increases operational complexity.
+
+## Proposals
+
+### Proposal 0: Do nothing
+
+Loki's current `ruler` implementation is sufficient for small installations running relatively simple or inexpensive queries.
+
+**Pros**:
+- Nothing to be done
+
+**Cons:**
+- Loki's `ruler` will remain unreliable and inefficient when used in large multi-tenant environments with expensive queries.
+
+### Proposal 1: Remote Execution
```

Comment:
IIRC, we discussed the pros and cons of two ways of achieving remote execution:
1. Re-use the existing frontend+scheduler+queriers
2. Create a separate set of frontend+scheduler+queriers

I think we did not discard any of those two alternatives, so I think it may be worth adding a third proposal with (2.).

Reply:
The workload is referring to user queries. I'll add some more detail here to improve the clarity, thanks!

Reply:
See https://github.com/grafana/loki/pull/8129/commits/1aad3e950eca4289a92b32f10c97e5484c32b822